lib/charms/opensearch/v0/opensearch_keystore.py

```diff
@@ -97,5 +122,91 @@ def _reload_keystore(self) -> None:
             raise OpenSearchKeystoreError(
                 f"Failed to reload keystore: error code: {e.response_code}, error body: {e.response_body}"
             )
-        if response.get("status", 400) >= 400:
-            raise OpenSearchKeystoreError("Failed to reload keystore with: {response}")
+
+
+class OpenSearchCATruststore(Keystore):
+    """Manages the default CA truststore."""
+
+    def __init__(self, charm):
+        """Creates the keystore manager class."""
+        super().__init__(charm)
+        self._keystore = charm.opensearch.paths.ca_truststore
+        self._password = "changeit"
+
+    def list(self, alias: str = None) -> List[str]:
+        """Lists the keys available in opensearch's keystore."""
+        try:
+            # Not using OPENSEARCH_BIN path
+            return self._opensearch._run_cmd(self._keytool, f"-v -list -keystore {self._keystore}")
+        except OpenSearchCmdError as e:
+            raise OpenSearchKeystoreError(str(e))
+
+    def add(self, entries: Dict[str, str]) -> None:
+        """Adds a new set of entries to the keystore."""
+        if not entries:
+            raise OpenSearchKeystoreError("Missing entries for keystore")
+        for key, value in entries.items():
+            self._add(key, value)
```

Comment:
the `_add` method content can  be put here instead of having it as a separate method.

Reply:
I use the _add for two different tasks: adding a value to the store, and adding a cert file